control/src/nodelets/mushr_feedback_controller.cpp

```diff
@@ -0,0 +1,165 @@
+    plant->get_control_space()->set_bounds(min_control_limits, max_control_limits);
+
+    // prx::world_model_t control_model({ plant }, {});
+    std::vector<prx::system_ptr_t> systems = { plant };
+    std::vector<std::shared_ptr<prx::movable_object_t>> obstacles = {};
+    control_model = std::make_shared<prx::world_model_t>(systems, obstacles);
+    control_model->create_context("control_context", { plant_name }, {});
+    context = control_model->get_context("control_context");
+    ROS_ASSERT(context.first != nullptr);
+
+    trajectory = std::make_shared<prx::trajectory_t>(plant->get_state_space());
+    current_state = plant->get_state_space()->make_point();
+
+    if (controller_params["name"].as<std::string>() == "mushr_pure_pursuit")
+    {
+      controller = new prx::mushr_pure_pursuit_t(plant, controller_params);
+    }
+    else if (controller_params["name"].as<std::string>() == "mushr_stanley")
+    {
+      controller = new prx::mushr_stanley_t(plant, controller_params);
+    }
+    else
+    {
+      ROS_ERROR("Unknown controller type: %s", controller_params["name"].as<std::string>().c_str());
+      return;
+    }
+    controller->set_goal(ml4kp_params["goal_state"].as<std::vector<double>>());
+  }
+
+  void onInit()
+  {
+    ros::NodeHandle& nh{ getPrivateNodeHandle() };
+    std::string trajectory_topic, pose_topic, control_topic, reset_topic;
+
+    nh.getParam("trajectory_topic", trajectory_topic);
+    nh.getParam("pose_topic", pose_topic);
+    nh.getParam("control_topic", control_topic);
+    nh.getParam("reset_topic", reset_topic);
+
+    ros::Duration _timer_duration = ros::Duration(controller->control_duration());
+    _timer = nh.createTimer(_timer_duration, &mushr_feedback_controller_t::control, this);
+
+    _trajectory_subscriber = nh.subscribe(trajectory_topic, 1, &mushr_feedback_controller_t::get_trajectory, this);
+    _pose_subscriber = nh.subscribe(pose_topic, 1, &mushr_feedback_controller_t::get_pose, this);
+    _reset_subscriber = nh.subscribe(reset_topic, 1, &mushr_feedback_controller_t::reset, this);
+
+    _control_publisher = nh.advertise<ml4kp_bridge::SpacePoint>(control_topic, 1, true);
+    _ctrl.point.resize(plant->get_control_space()->get_dimension());
+
+    _service_server =
+        nh.advertiseService("/mushr/control_propagation", &mushr_feedback_controller_t::control_propagation, this);
+  }
+
+  void get_trajectory(const ml4kp_bridge::TrajectoryStamped& msg)
+  {
+    ml4kp_bridge::copy(trajectory, msg.trajectory);
+    controller->set_points(trajectory);
+    _trajectory_received = true;
+  }
+
+  void get_pose(const prx_models::MushrObservation& msg)
+  {
+    prx_models::copy(current_state, msg);
+    _pose_received = true;
+  }
+
+  void reset(const std_msgs::Empty& msg)
+  {
+    _trajectory_received = false;
+    _pose_received = false;
+    controller->reset();
+  }
+
+  void control(const ros::TimerEvent& event)
+  {
+    controller->get_control(current_state, current_control);
```

Comment:
`get_control` is only defined for `mushr_pure_persuit`, not for the feedback controller.